Sources/NetworkProtection/KeyManagement/NetworkProtectionKeyStore.swift

```diff
@@ -138,13 +126,13 @@ public final class NetworkProtectionKeychainKeyStore: NetworkProtectionKeyStore
     }
 
     public func updateKeyPair(_ newKeyPair: KeyPair) {
-        self.currentPrivateKey = newKeyPair.privateKey
-        self.currentExpirationDate = newKeyPair.expirationDate
-    }
+        if currentPrivateKey != newKeyPair.privateKey {
```

Comment:
Make sure we don't try to overwritte things that haven't changed.

Reply:
Do you think we should update the expiration date anyway, to be safe? In the case that a new key gets set here, which has a different expiration date from what this class knows about.